Automattic/jetpack
# Track/block type for memberships

Fixes 181-gh-automattic/gold

## Proposed changes:

* Adds a new attribute `block_type` to each URL for the memberships checkout form iframe to be passed along for segmentation purposes.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up a new Jurassic Ninja site with Jetpack beta and activate this feature branch
* Setup Jetpack constants and enable global-http
* Sandbox the public api, subscribe.wordpress.com and enable store sandbox
* Sandbox all async jobs and open the watcher on your sandbox.
* Apply D127463-code to your sandbox
* Download this branch to your sandbox on top of D127463-code (answer "n" then "y").
* Connect your JN site to Jetpack, then activate subscriptions and connect to Stripe.
* Setup a paid content block with a plan.
* Setup a paywall block on another post.
* Setup a payment button on another post.
* Setup a donate form on another post.
* On your sandbox, tail php-errors
* Open a private window and mouse over the subscribe button for each block - you should see a `block_type=` attribute which matches the block you are clicking on.
* Subscribe to each of the different payment blocks on the posts you created.
* You should see confirmation of fired events with `block_type` for the `wpcom_memberships_checkout_accessed` event in php-errors.
